### PR TITLE
feat: Add 2d selection to XYAxis

### PR DESCRIPTION
**Pull request by jbrundage**
@GordonSmith please review

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

### CODE REVIEW

File: packages/chart/src/XYAxis.ts [hunk 1]
```diff
@@ -183,23 +192,37 @@ export class XYAxis extends SVGWidget {
         let selected = [];
         const currSel: any = d3BrushSelection(this.svgBrush.node());
         if (currSel) {
-            selected = this.data().filter(function (d) {
-                let pos = d[0];
-                pos = this.dataPos(pos) + this.domainAxis.bandwidth() / 2;
-                return pos >= currSel[0] && pos <= currSel[1];
-            }, this);
+            if (this.selectionMode2D()) {
+                selected = this.data().filter(function (d) {
```
Comment (GordonSmith) [inline packages/chart/src/XYAxis.ts]: Minor:  I would probably put the `if (this.selectionMode2D()) {` inside the `selected = this.data().filter(function (d) {` and reuse some of the "pos calculations.

File: packages/chart/src/XYAxis.ts [hunk 2]
```diff
@@ -376,16 +399,21 @@ export class XYAxis extends SVGWidget {
...
             ;
-
-        const handlePath = this.svgBrush.selectAll(".handle--custom").data(isHorizontal ? [{ type: "w" }, { type: "e" }] : [{ type: "n" }, { type: "s" }]);
+        let handleTypes;
+        if (this.selectionMode2D()) {
+            handleTypes = [];
+        } else {
+            handleTypes = isHorizontal ? [{ type: "w" }, { type: "e" }] : [{ type: "n" }, { type: "s" }];
+        }
+        const handlePath = this.svgBrush.selectAll(".handle--custom").data(handleTypes);
```
Comment (GordonSmith) [inline packages/chart/src/XYAxis.ts]: Duplicate code to 214->220.

File: packages/chart/src/XYAxis.ts [hunk 3]
```diff
@@ -580,6 +608,8 @@ export class XYAxis extends SVGWidget {
     orientation: publish<this, string>;
     orientation_default: publish<this, string>;
 
+    @publish(false, "boolean", "2D Range Selector")
+    selectionMode2D: publish<this, boolean>;
     @publish(false, "boolean", "Range Selector")
     selectionMode: publish<this, boolean>;
```
Comment (GordonSmith) [inline packages/chart/src/XYAxis.ts]: Should we have a single PP for both of these?  "selectionMode" with options "1D", "2D"?

Comment (jbrundage) [reply packages/chart/src/XYAxis.ts]: I thought about this but didn't want to make this a potentially breaking change.

**Comment by GordonSmith** [comment]
@jbrundage can you rebase onto latest?

**Pull request closed without merging**